web/src/utils/schema/editor.ts

```diff
@@ -0,0 +1,80 @@
+import type { JSONSchema7 } from 'json-schema';
+
+import Vue from 'vue';
+import {
+  computed, reactive, ref, ComputedRef,
+} from '@vue/composition-api';
+import { cloneDeep } from 'lodash';
+
+import { DandiModel } from './types';
+import {
+  computeBasicSchema,
+  computeComplexSchema,
+  filterModelWithSchema,
+  writeSubModelToMaster,
+  populateEmptyArrays,
+} from './utils';
+
+/**
+ * Manages the interface between the source data/schema, and the changes necessary for it to
+ * operate correctly with the Meditor.
+ */
+class EditorInterface {
+  // Not guaranteed to be up to date, use getModel()
+  private model: DandiModel;
+
+  basicModel: DandiModel;
+  complexModel: DandiModel;
```

Comment:
It's not two different modes, but the same mode split up and used in several places. If the `VJSF` component was not split up, and rendered the whole schema at once, it would get pretty unwieldy pretty fast, since there's a lot of data that can exist on a dandiset, it gets hard to see the when one section starts vs. ends, etc. To get around that, the schema + model are split into two different sections: The "basic" schema + model, and the "complex" schema + model.

The basic schema is rendered normally like the original would have been, but these fields are all simple or "basic" types, like strings, numbers, booleans, etc., as well as arrays of these types (since in the component arrays of something like strings is just rendered as a multi-selection vuetify form).

The complex schema is all other schema fields that aren't captured by the basic schema. Each of these complex fields are rendered individually (with an individual `VJSF` component) underneath those button dialogs at the top of the meditor.

Hopefully this clears things up.